.github/actions/spelling/patterns.txt

```diff
@@ -1,38 +1,86 @@
 # See https://github.com/check-spelling/check-spelling/wiki/Configuration-Examples:-patterns
 
-# stackexchange -- https://stackexchange.com/feeds/sites
-\b(?:askubuntu|serverfault|stack(?:exchange|overflow)|superuser).com/questions/\d+/[a-z-]+
-# w3
-\bw3\.org/[-0-9a-zA-Z/#.]+
-# mvnrepository.com
-\bmvnrepository\.com/[-0-9a-z./]+
-# URL escaped characters
-\%[0-9A-F]{2}
-# sha-1
-"[0-9a-f]{40}"
+# Automatically suggested patterns
+# hit-count: 145 file-count: 14
 # hex digits including css/html color classes:
-(?:[\\0][xX]|\\u|[uU]\+|#|\%23)[0-9a-fA-FgGrR]{2,}[uU]?[lL]{0,2}\b
+(?:[\\0][xX]|\\u|[uU]\+|#x?|\%23)[0-9_a-fA-FgGrR]*?[a-fA-FgGrR]{2,}[0-9_a-fA-FgGrR]*(?:[uUlL]{0,3}|[iu]\d+)\b
+
+# hit-count: 14 file-count: 11
+# https/http/file urls
+(?:\b(?:https?|ftp|file)://)[-A-Za-z0-9+&@#/%?=~_|!:,.;]+[-A-Za-z0-9+&@#/%=~_|]
+
+# hit-count: 13 file-count: 4
+# in check-spelling@v0.0.22+, printf markers aren't automatically consumed
+# printf markers
+(?<!\\)\\[nrt](?=[a-z]{2,})
+
+# hit-count: 11 file-count: 4
+# base64 encoded content
+([`'"])[-a-zA-Z=;:/0-9+]+?[0-9]+?[-a-zA-Z=;:/0-9+]+=\g{-1}
+
+# hit-count: 10 file-count: 8
+# IServiceProvider / isAThing
+\b(?:I|isA)(?=(?:[A-Z][a-z]{2,})+(?:[A-Z]|\b))
+
+# hit-count: 4 file-count: 3
+# hex runs
+\b[0-9a-fA-F]{16,}\b
+
+# hit-count: 4 file-count: 2
 # uuid:
-[{"'][0-9a-fA-F]{8}-(?:[0-9a-fA-F]{4}-){3}[0-9a-fA-F]{12}['"}]
-# curl arguments
-\b(?:)curl(?:\s+-[a-zA-Z]+)+
-# tar arguments
-\b(?:)tar(?:\s+-[a-zA-Z]+|\s[a-z]+)+
+\b[0-9a-fA-F]{8}-(?:[0-9a-fA-F]{4}-){3}[0-9a-fA-F]{12}\b
+
+# hit-count: 2 file-count: 1
+# sha-... -- uses a fancy capture
+(\\?['"]|&quot;)[0-9a-f]{40,}\g{-1}
+
+# hit-count: 1 file-count: 1
+# Time Zones
+\b(?:Africa|Atlantic|America|Antarctica|Asia|Australia|Europe|Indian|Pacific)(?:/\w+)+
+
 # decode
 \.decode\("[0-9a-zA-Z]+"
-#
-"[0-9a-f]{80,}"
+
 # pom.xml
-<(additionalClasspathElement|arg|(?:artifact|group)Id|id|installDir|jvmArgs|mainClass|shadedClassifierName|template(?:Base|Source|Output)Dir|version)>[^<]*</\1>
+<(additionalClasspathElement|arg|(?:artifact|group)Id|id|installDir|jvmArgs|mainClass|shadedClassifierName|template(?:Base|Source|Output)Dir|version)>[^<]*</\g{-1}>
```

Comment:
the `\1` notation is very fragile, the `\g{-1}` notation will find the most recent match instead of the first match (the `(\\?['"]|&quot;)[0-9a-f]{40,}\g{-1}` rule above shifted what `\1` referenced)